CMakeLists.txt

```diff
@@ -748,15 +745,51 @@ if(BUSTED_LUA_PRG)
     set_target_properties(functionaltest-lua PROPERTIES FOLDER test)
 endif()
 
-if(LUACHECK_PRG)
-  add_custom_target(lualint
-    COMMAND ${LUACHECK_PRG} -q runtime/ scripts/ src/ test/
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-else()
-  add_custom_target(lualint false
-    COMMENT "lualint: LUACHECK_PRG not defined")
-endif()
-
+find_program(LUACHECK luacheck)
+find_program(STYLUA stylua)
+find_program(SHELLCHECK shellcheck)
+find_program(FLAKE8 flake8)
+find_program(UNCRUSTIFY uncrustify)
+
+set(TARGET "lualint")
+add_custom_target(${TARGET}
+  COMMAND ${CMAKE_COMMAND}
+  -DPROGRAM=${LUACHECK}
+  -DPROJECT_ROOT=${PROJECT_SOURCE_DIR}
+  -DTARGET=${TARGET}
+  -P ${PROJECT_SOURCE_DIR}/cmake/lint.cmake)
+
+set(TARGET "stylua")
```

Comment:
all lint tasks should have `lint` suffix. Or better, a `lint` prefix, since we're changing things already.